### DISCUSSION TITLE
Question about the clz builtin

### DISCUSSION THREAD

**Discussion by tdavidcl** [AdaptiveCpp/AdaptiveCpp#724] [Q&A] [answered]
Hello,
I'm trying to support HipSYCL as compiler for an HPC code. I only struggle with one thing : 
my code is using an algorithm based on the use of the clz builtin which isn't implemented currently in hipSYCL. 
By any chance do you know any workaround ?

**Comment by illuhad** [comment, marked as answer]
You could either
- just implement it (probably not very difficult)
- leverage that hipSYCL allows you to embed code from other programming models such as CUDA or HIP into your SYCL kernels. This is effectively also how those builtins are implemented in hipSYCL. https://github.com/illuhad/hipSYCL/blob/develop/doc/hip-source-interop.md
